File: openapi-test-scala-server-http4s-client-http4s/testapi/api/PetsApi.scala
package testapi.api

import cats.effect.IO
import io.circe.Json
import java.lang.Void
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate

trait PetsApi {
  /** Create a pet */
  def createPet(body: PetCreate): IO[Response201400[Pet, Error]]

  /** Delete a pet */
  def deletePet(
    /** The pet ID */
    petId: String
  ): IO[Response404Default[Error]]

  /** Get a pet by ID */
  def getPet(
    /** The pet ID */
    petId: String
  ): IO[Response200404[Pet, Error]]

  /** Get pet photo */
  def getPetPhoto(
    /** The pet ID */
    petId: String
  ): IO[Void]

  /** List all pets */
  def listPets(
    /** Maximum number of pets to return */
    limit: Option[Int],
    /** Filter by status */
    status: Option[String]
  ): IO[List[Pet]]

  /** Upload a pet photo */
  def uploadPetPhoto(
    /** The pet ID */
    petId: String,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array[Byte]
  ): IO[Json]
}

File: openapi-test-java-server-jaxrs-client-jdk/testapi/api/PetsApi.java
package testapi.api;

import com.fasterxml.jackson.databind.JsonNode;
import java.lang.Void;
import java.util.List;
import java.util.Optional;
import testapi.model.Error;
import testapi.model.Pet;
import testapi.model.PetCreate;
import testapi.model.PetId;

public interface PetsApi {
  /** Create a pet */
  Response201400<Pet, Error> createPet(PetCreate body) throws java.lang.Exception;

  /** Delete a pet */
  Void deletePet(
  
    /** The pet ID */
    PetId petId
  ) throws java.lang.Exception;

  /** Get a pet by ID */
  Response200404<Pet, Error> getPet(
  
    /** The pet ID */
    PetId petId
  ) throws java.lang.Exception;

  /** Get pet photo */
  Void getPetPhoto(
  
    /** The pet ID */
    PetId petId
  ) throws java.lang.Exception;

  /** List all pets */
  List<Pet> listPets(
    /** Maximum number of pets to return */
    Optional<Integer> limit,
    /** Filter by status */
    Optional<String> status
  ) throws java.lang.Exception;

  /** Upload a pet photo */
  JsonNode uploadPetPhoto(
    /** The pet ID */
    PetId petId,
    /** Optional caption for the photo */
    String caption,
    /** The photo file to upload */
    Byte[] file
  ) throws java.lang.Exception;
}